File: spare_parts_app_new/lib/models/category.dart
class Category {
  final int id;
  final String name;
  final String? description;
  final String? imagePath;
  final String? imageLink;
  final int displayOrder;
  final int? iconCodePoint;
  final bool showOnHome;
  final bool deleted;

  Category({
    required this.id,
    required this.name,
    this.description,
    this.imagePath,
    this.imageLink,
    this.displayOrder = 0,
    this.iconCodePoint,
    this.showOnHome = true,
    this.deleted = false,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] is int ? json['id'] : (json['id'] as num?)?.toInt() ?? 0,
      name: json['name'] ?? '',
      description: json['description'],
      imagePath: json['imagePath'],
      imageLink: json['imageLink'],
      displayOrder: (json['displayOrder'] as num? ?? 0).toInt(),
      iconCodePoint: json['iconCodePoint'] != null ? (json['iconCodePoint'] as num).toInt() : null,
      showOnHome: json['showOnHome'] == true || json['showOnHome'] == 1,
      deleted: json['deleted'] == true || json['deleted'] == 1,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'imagePath': imagePath,
      'imageLink': imageLink,
      'displayOrder': displayOrder,
      'iconCodePoint': iconCodePoint,
      'showOnHome': showOnHome ? 1 : 0,
      'deleted': deleted,
    };
  }
}

File: backend/src/main/java/com/spareparts/inventory/entity/Category.java
package com.spareparts.inventory.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "categories")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@EqualsAndHashCode
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 120)
    private String name;

    @Column(length = 500)
    private String description;

    @Column(length = 500)
    private String imagePath;

    @Column(length = 500)
    private String imageLink;

    @Column(name = "display_order", columnDefinition = "integer default 0")
    private Integer displayOrder = 0;

    @Column(name = "icon_code_point")
    private Integer iconCodePoint;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
    
    @Column(name = "deleted", nullable = false, columnDefinition = "boolean default false")
    private boolean deleted = false;
}
